src/task.c

```diff
@@ -86,7 +91,10 @@ task_init_threadid(task_system *ts)
    }
 
 out:
-   debug_assert(tid != INVALID_TID);
+   platform_assert((tid != INVALID_TID),
```

Comment:
Now, we will get an error in debug & release builds. Have a new unit-test case that triggers this error.

I am reworking that case slightly to come up with a scenario where bg-threads use up all but-one-slot. Then, we should be able to create one more user thread and the next create should fail. This small test-case dev is under progress, while we speak ...

Reply:
Update: Uploaded new commit with enhanced test case `test_use_all_but_one_threads_for_bg_threads()`.